airflow/providers/amazon/aws/sensors/dynamodb.py

```diff
@@ -0,0 +1,96 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import os
+import boto3
+from pprint import pprint
+from botocore.exceptions import ClientError
+
+from airflow.sensors.base import BaseSensorOperator
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class DynamoDBSensor(BaseSensorOperator):
+    """
+    Waits for an attribute value to be present for an item in a DynamoDB table.
+
+    :param partition_key_name: DynamoDB partition key name
+    :param partition_key_value: DynamoDB partition key value
+    :param attribute_name: DynamoDB attribute name
+    :param attribute_value: DynamoDB attribute value
+    :param sort_key_name: (optional) DynamoDB sort key name
+    :param sort_key_value: (optional) DynamoDB sort key value
+    """
+
+    def __init__(
+        self,
+        table_name: str,
+        partition_key_name=None,
+        partition_key_value=None,
+        attribute_name=None,
+        attribute_value=None,
+        sort_key_name: Optional[str] = None,
+        sort_key_value: Optional[str] = None,
```

Comment:
how about taking those two as a tuple, considering they need to be specified or omitted together ?

Reply:
Not sure I like this, since I don't see it very often in Airflow, and it requires packing/unpacking which might be cumbersome and unintuitive. I'd prefer to simply raise an exception if the pair is not supplied. What do you think?

Reply:
ok makes sense, let's continue the conversation on the public PR